client/service/src/client/client.rs

```diff
@@ -1047,6 +1085,38 @@ where
 		Ok(())
 	}
 
+	fn notify_imported_for_every_block(
+		&self,
+		notification: Option<BlockImportNotification<Block>>,
+		storage_changes: Option<(StorageCollection, ChildStorageCollection)>,
+	) -> sp_blockchain::Result<()> {
```

Comment:
We should not clone this method. You should use the old one and forward the `import_notification_action` and then do the "right" thing here internally.